embedder.py

```diff
@@ -0,0 +1,152 @@
+logging.basicConfig(format="%(levelname)s - %(asctime)s: %(message)s", datefmt= '%H:%M:%S', level=logging.INFO)
+
+
+#CONFIG
+OUTPUT_DIRECTORY = os.path.join('./output')
+PARAM_DF_PATH = os.path.join(OUTPUT_DIRECTORY, 'ml_inputs', '_ml_param.csv')
+RETURN_DF_PATH = os.path.join(OUTPUT_DIRECTORY, 'ml_inputs', '_ml_return.csv')
+OUTPUT_EMBEDDINGS_DIRECTORY = os.path.join('./resources')
+LANGUAGE_EMBEDDING_OUTPUT_PATH = os.path.join(OUTPUT_EMBEDDINGS_DIRECTORY, 'w2v_language_model.bin')
+CODE_EMBEDDING_OUTPUT_PATH = os.path.join(OUTPUT_EMBEDDINGS_DIRECTORY, 'w2v_code_model.bin')
+
+
+
+class HelperIterator:
+    """
+    Subclass for type Hinting the iterators listed below
+    """
+    pass
+
+
+class LanguageIterator(HelperIterator):
+    """
+    Helper Iterator that iterates over the whole collection of descriptions language.
+    """
+    def __init__(self, param_df: pd.DataFrame, return_df: pd.DataFrame) -> None:
+        self.param_df = param_df
+        self.return_df = return_df
+
+    def __iter__(self):
+        for func_descr_sentence in self.return_df['func_descr']:
+            yield func_descr_sentence.split()
+
+        for param_descr_sentence in self.param_df['arg_comment']:
+            yield  param_descr_sentence.split()
+
+        for return_descr_sentence in self.return_df['return_descr']:
+            yield return_descr_sentence.split()
+
+
+class CodeIterator(HelperIterator):
+    """
+    Helper Iterator that iterates over the whole collection of the code expressions.
+    """
+    def __init__(self, param_df: pd.DataFrame, return_df: pd.DataFrame) -> None:
+        self.param_df = param_df
+        self.return_df = return_df
+
+    def __iter__(self):
+        for return_expr_sentences in self.return_df['return_expr_str']:
+            yield return_expr_sentences.split()
+
+        for func_name_sentences in self.return_df['name']:
+            yield func_name_sentences.split()
+
+        for arg_names_sentences in self.return_df['arg_names_str']:
+            yield arg_names_sentences.split()
+
+
+class Embedder:
+    """
+    Create embeddings for the code names and docstring names using Word2Vec.
+    """
+
+    def __init__(self, param_df: pd.DataFrame, return_df: pd.DataFrame) -> None:
+        self.param_df = param_df
+        self.return_df = return_df
+
+    def trainModel(self, corpus_iterator: HelperIterator, model_path_name: str) -> None:
+        """
+        Train a Word2Vec model and save the output to a file.
+        :param corpus_iterator: class that can provide an iterator that goes through the corpus
+        :param model_path_name: path name of the output file
+        """
+
+        cores = multiprocessing.cpu_count()
+
+        w2v_model = Word2Vec(min_count=5,           #Specified in NL2Type
+                             #window=2,
+                             size=100,              #Specified in NL2Type
+                             #sample=6e-5,
```

Comment:
Do we want to leave this parameters here if we don't use them?

Reply:
I guess they can be added back when we are testing different hyperparameters settings.